src/pydirectory/Directory.py

```diff
@@ -175,30 +182,30 @@ def __init__(self, directory, path='', name=None, pathName=None,
 
     # ----- Name and Path setters
     @property
-    def path(self):
+    def path(self) -> str:
         return os.path.dirname(self.pathName)
 
     @path.setter
-    def path(self, path):
+    def path(self, path: str):
         path = path if path else ''
         self.pathName = os.path.join(path, self.name)
 
     @property
-    def name(self):
+    def name(self) -> str:
         return os.path.basename(self.pathName)
 
     @name.setter
-    def name(self, name):
+    def name(self, name: str):
         self.pathName = os.path.join(self.path, name)
 
     # ----- pathName functions
 
     @property
-    def pathName(self):
+    def pathName(self) -> str:
         return self._pathName
 
     @pathName.setter
-    def pathName(self, pathName):
+    def pathName(self, pathName: str):
         if self.pathName == pathName:
             return
```

Comment:
I set `pathName` type to `str` because or the previous definition:
`    @property
    def pathName(self) -> str:
        return self._pathName

    @pathName.settergit 
    def pathName(self, pathName: str | None):
        if self.pathName == pathName:
            return`

I left off `-> ` because return is none